public/app/features/query/state/runRequest.ts

```diff
@@ -90,6 +83,20 @@ export function processResponsePacket(packet: DataQueryResponse, state: RunningQ
   return { packets, panelData };
 }
 
+function getRequestTimeRange(request: DataQueryRequest, loadingState: LoadingState): TimeRange {
+  const range = request.range;
+
+  if (!isString(range.raw.from) || loadingState !== LoadingState.Streaming) {
+    return range;
+  }
+
+  return {
+    ...range,
+    from: dateMath.parse(range.raw.from, false)!,
+    to: dateMath.parse(range.raw.to, true)!,
+  };
+}
```

Comment:
So this changes logic to always use the request range as originally passed in, and leaves the time range update & re-evaluation to only on streaming